propnet/ext/matproj.py

```diff
@@ -22,104 +22,70 @@
     'final_energy_per_atom': 'final_energy_per_atom',
     'formation_energy_per_atom': 'formation_energy_per_atom',
     'piezoelectric_tensor': 'piezo.piezoelectric_tensor',
-    'volume_unit_cell': 'volume',
-    'structure': 'structure'
+    'volume_unit_cell': 'volume'
 }
 PROPNET_FROM_MP_NAME_MAPPING = {v: k for k, v in MP_FROM_PROPNET_NAME_MAPPING.items()}
 
 # list of all available properties
-AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id']
+AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id', 'structure']
```

Comment:
Why are these two items being added here, instead of adding them to the `PROPNET_FROM_MP_NAME_MAPPING` ?  If there's a reason they're added here, please add a comment.

(Aside: 'task_id' is a bit of a strange one ... perhaps we can map that to some kind of 'identifier' object?)

Reply:
I just put the task id in tags for now?

structure and task_id are added separately because they aren't handled the same way the dictionary items are handled.

- Dictionary entries have a 1:1 mapping to Symbols attached to the returned material.
- task_id is not currently mapped to a Symbol it's stored in tags.
- structure has a 1:2 mapping -- it is stored in a Structure Symbol and then a Lattice Symbol.

So I didn't add task_id or structure to the dictionary because it kept the code easier to write. We could add an if-statement and check for them in the loop? In that case we could add them to the dictionary, but we don't have an explicit symbol for task_id either, so I'm not sure what it would map to.